src/ast/rewriter/seq_rewriter.cpp

```diff
@@ -2056,6 +2367,22 @@ br_status seq_rewriter::mk_str_in_regexp(expr* a, expr* b, expr_ref& result) {
         result = m().mk_eq(a, b1);
         return BR_REWRITE1;
     }
+    if (m_util.str.is_empty(a)) {
+        result = is_nullable(b);
+        return BR_REWRITE_FULL; // is_nullable doesn't rewrite
```

Comment:
should return BR_DONE.
If is_nullable returns (str.in_re s r), it could not rewrite it any more. Returning anything but BR_DONE or BR_FAILED would reinvoke rewriting leading to a loop

Reply:
Yes, there was a loop here. But BR_DONE assumes is_nullable rewrites fully. What about the following?
    if (m_util.str.is_empty(a)) {
        result = is_nullable(b);
        // is_nullable doesn't rewrite fully. But we also want to avoid the
        // case where it didn't succeed in changing anything.
        if (m_util.str.is_in_re(result)) {
            return BR_FAILED;
        }
        else {
            return BR_REWRITE_FULL;
        }
    }

Reply:
sure